tamr_client/dataset/_dataset.py

```diff
@@ -262,5 +262,5 @@ def create(
     data = response.successful(r).json()
     dataset_path = data["relativeId"]
     dataset_url = URL(instance=instance, path=str(dataset_path))
-
+    # There may be a slight perf hit from using a GET to construct the dataset but this is OK
```

Comment:
I think this sort of comment belongs more in the `Contributor Docs` that inline given that we use this pattern (i.e. "Don't be fancy about avoiding extra requests until we identify a perf issue") throughout the codebase.

Reply:
Does this seem like an ADR-level note, or does it belong somewhere else in the Contributor Docs?

Reply:
Yea, actually ADR seems like the right place. Good point.